GraphBulkImporter/Extensions.cs

```diff
@@ -0,0 +1,103 @@
+        /// </returns>
+        public static GremlinVertex ToGremlinVertex(this object obj)
+        {
+            var label = obj.GetType().Name;
+            
+            if (!obj.HasProperty("id"))
+            {
+                throw new ArgumentException($"{label} does not have expected property.", "id");
+            }
+
+            if (!obj.HasProperty("partitionKey"))
+            {
+                throw new ArgumentException($"{label} does not have expected property.", "partitionKey");
+            }
+
+            return ToGremlinVertex(obj, "id", "partitionKey", label);
+        }
+
+        /// <summary>
+        /// Converts an object to a GremlinVertex allowing more control over how the id, partitionKey, and label properties are set. 
+        /// </summary>
+        /// <param name="idProperty">Which property of the object should be used for id of the GremlinVertex, defaults to "id"</param>
+        /// <param name="partitionKeyProperty">Which property should be used for partitionKey of the GremlinVertex, defaults to "partitionKey"</param>
+        /// <param name="label">What value should be used for the label property, defaults to the Type name of the object</param>
+        /// <returns>
+        /// A new instance of a GremlinVertex
+        /// with its id and partitionKey property values set to and <typeparamref name="idProperty"/> and <typeparamref name="partitionKeyProeprty"/> properties respectively
+        /// and the label property set to the value of <typeparamref name="label"/>
+        /// </returns>
+        public static GremlinVertex ToGremlinVertex(this object obj, string idProperty,  string partitionKeyProperty, string label)
+        {
+            if (string.IsNullOrEmpty(idProperty))
+            {
+                throw new ArgumentException($"{nameof(idProperty)} cannot be Null or Empty", nameof(idProperty));
+            }
+
+            if (string.IsNullOrWhiteSpace(partitionKeyProperty))
+            {
+                throw new ArgumentException($"{nameof(partitionKeyProperty)} cannot be Null or Empty", nameof(partitionKeyProperty));
+            }
+
+            if (string.IsNullOrWhiteSpace(label))
+            {
+                throw new ArgumentException($"{nameof(label)} cannot be Null or Empty", nameof(label));
+            }
+
+            var gv = new GremlinVertex(obj.GetPropertyValue(idProperty).ToString(), label);
+            gv.AddProperty(new GremlinVertexProperty("partitionKey", obj.GetPropertyValue(partitionKeyProperty)));
+
+            foreach (var prop in obj.GetType().GetProperties(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance))
+            {
+                var name = prop.Name;
+                var val = prop.GetValue(obj);
+
+                if (!name.Equals("id", StringComparison.InvariantCultureIgnoreCase) && !name.Equals("partitionKey", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
instead of checking this in the loop each time we can remove these properties from the list of properties before we start the loop, something like this:
```obj.GetType().GetProperties(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance).Where(x => !x.Name.Equals("id", StringComparison.InvariantCultureIgnoreCase) && !x.Name.Equals("partitionKey", StringComparison.InvariantCultureIgnoreCase))```

Reply:
good point. 
changed to - 

```
            var props = obj.GetType().GetProperties()
                .Where(x => !x.Name.Equals("id", StringComparison.InvariantCultureIgnoreCase)
                && !x.Name.Equals("partitionKey", StringComparison.InvariantCultureIgnoreCase));

            foreach (var prop in props) {           
                gv.AddProperty(new GremlinVertexProperty(prop.Name, prop.GetValue(obj)));
            }
```